plugins/modules/system/puppet.py

```diff
@@ -268,8 +269,15 @@ def main():
         cmd = "%s apply --detailed-exitcodes " % base_cmd
         if p['logdest'] == 'syslog':
             cmd += "--logdest syslog "
+        if p['logdest'] == 'stdout':
+            cmd += "--logdest console "
+            module.warn(
```

Comment:
Use `module.deprecate()` for deprecation messages.

I think it's better to move the deprecation into a separate PR, and instead just mention that `stdout` will be deprecated eventually. This also gives users a better upgrade path.

Reply:
I was going to, but I wasn't sure what to do here, we use the word "deprecate" here over and over, when we are in fact fixing a bug. That's why I used `3.0.1` here and there, and `module.warn` instead of `module.deprecate`.

Reply:
I removed the warning, and if a user has explicitly stated `stdout` in their task, they won't be affected for now.
https://github.com/ansible-collections/community.general/pull/2407/commits/7b1428f70a4e7b5d6b4ee34f5b835ac725c0fb80

Although I'm still not sure why we can't do this all in `3.0.1` as this is a bug.

Reply:
We were mainly talking about deprecation because you wanted to remove the old value :) For a bugfix, it's best to only fix the bug and not do other changes. Strictly speaking, this PR is adding a new feature, namely a new value in `choices` for `log_dest`.

For this to be a proper bugfix that can be backported, only the code which handles `if p['logdest'] == 'all':` should be touched.

Reply:
I'd really appreciate if you could take a new look at this PR as a bugfix.
I will open a new PR after this has been merged deprecating `stdout` in c.g. 3.1.0.

Reply:
It's probably best to first mention it will be deprecated, and then deprecate it in 4.0.0. This gives users some time where the old value still works without triggering deprecation warnings.